domain/usecases/subscribe_test.go

```diff
@@ -0,0 +1,146 @@
+package usecases
+
+import (
+	"context"
+	"testing"
+
+	"github.com/matheusmosca/walrus/domain/entities"
+	"github.com/matheusmosca/walrus/domain/vos"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSubsribe(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		ctx       context.Context
+		topicName vos.TopicName
+	}
+
+	type fields struct {
+		storage   Repository
+		topicName vos.TopicName
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		fields  fields
+		wantCh  chan vos.Message
+		wantId  vos.SubscriberID
+		wantErr error
+	}{
+		{
+			name: "subscribe should succeed",
+			args: args{
+				ctx:       context.Background(),
+				topicName: "walrus",
+			},
+			fields: fields{
+				storage:   &RepositoryMock{},
+				topicName: "walrus",
+			},
+			wantCh:  make(chan vos.Message),
+			wantId:  vos.SubscriberID("id"),
+			wantErr: nil,
+		},
+		{
+			name: "short topic name should return error",
+			args: args{
+				ctx:       context.Background(),
+				topicName: "xd",
+			},
+			fields: fields{
+				storage:   &RepositoryMock{},
+				topicName: "walrus",
+			},
+			wantCh:  nil,
+			wantId:  "",
+			wantErr: vos.ErrTopicNameTooShort,
+		},
+		{
+			name: "empty topic name message should return error",
+			args: args{
+				ctx:       context.Background(),
+				topicName: "",
+			},
+			fields: fields{
+				storage:   &RepositoryMock{},
+				topicName: "walrus",
+			},
+			wantCh:  nil,
+			wantId:  "",
+			wantErr: vos.ErrEmptyTopicName,
+		},
+		{
+			name: "nonexistent topic should return error",
```

Comment:
This scenario is wrong as well. If the topic does not exist, it should be created by Repository.CreateTopic().